bokehjs/gulp/tasks/test.coffee

```diff
@@ -12,71 +12,79 @@ mocha = (options={}) ->
     (file) ->
       if @_files? then @_files.push(file.path) else @_files = [file.path]
     () ->
-      if options.coverage? and options.coverage
-        args = [
-          "node_modules/.bin/istanbul",
-          "cover",
-          "node_modules/mocha/bin/_mocha",
-          "--",
-          "--compilers", "coffee:coffee-script/register",
-          "--reporter", argv.reporter ? "spec",
-          "./test/index.coffee"
-        ].concat(@_files)
+      _mocha = "node_modules/mocha/bin/_mocha"
+
+      if not options.coverage
+        args = [_mocha]
       else
-        args = [
-          "node_modules/mocha/bin/_mocha",
-          "--compilers", "coffee:coffee-script/register",
-          "--reporter", argv.reporter ? "spec",
-          "./test/index.coffee"
-        ].concat(@_files)
+        args = ["node_modules/.bin/istanbul", "cover", _mocha, "--"]
 
       if argv.debug
         args.unshift("debug")
 
-      proc = cp.spawn(process.execPath, args, {stdio: 'inherit'})
+      args = args.concat(
+        ["--compilers", "coffee:coffee-script/register,ts:ts-node/register"],
+        ["--reporter", argv.reporter ? "spec"],
+        ["--slow", "5s"]
+        ["./test/index.coffee"],
+        @_files,
+      )
+
+      env = Object.assign({}, process.env, {
+        TS_NODE_PROJECT: "./test/tsconfig.json"
+      })
+
+      proc = cp.spawn(process.execPath, args, {stdio: 'inherit', env: env})
 
       proc.on "error", (err) =>
         @emit("error", new gutil.PluginError("mocha", err))
 
-      proc.on "exit", (code) =>
+      proc.on "exit", (code, signal) =>
         if code != 0
-          @emit("error", new gutil.PluginError("mocha", "tests failed"))
+          comment = if signal == "SIGINT" then "interrupted" else "failed"
+          @emit("error", new gutil.PluginError("mocha", "tests #{comment}"))
         else
           @emit("end")
 
-      handler = () -> proc.kill()
-      process.on('exit',    handler)
-      process.on('SIGTERM', handler)
-      process.on('SIGINT',  handler)
+      process.on('exit',    () -> proc.kill())
+      process.on("SIGTERM", () -> proc.kill("SIGTERM"))
+      process.on("SIGINT",  () -> proc.kill("SIGINT"))
   )
 
-gulp.task "test:coverage", ["defaults:generate"], () ->
-  gulp.src(["./test/all.coffee"]).pipe(mocha({coverage: true}))
 
 gulp.task "test", ["defaults:generate"], () ->
-  gulp.src(["./test/all.coffee"]).pipe(mocha())
+  gulp.src(["./test/unit.coffee", "./test/defaults.coffee", "./test/size"]).pipe(mocha())
 
-gulp.task "test:client", ->
+gulp.task "test:unit", () ->
```

Comment:
`gulp test:unit` is now a way to run only unit tests, so nothing that depends on external factors (defaults, size).